source/Cosmos.System2/Graphics/SVGAIICanvas.cs

```diff
@@ -151,7 +172,20 @@ public override void DrawPoint(Pen aPen, float aX, float aY)
         /// <exception cref="NotImplementedException">Thrown if VMWare SVGA 2 has no rectange copy capability</exception>
         public override void DrawFilledRectangle(Pen aPen, int aX_start, int aY_start, int aWidth, int aHeight)
         {
-            _xSVGADriver.Fill((uint)aX_start, (uint)aY_start, (uint)aWidth, (uint)aHeight, (uint)aPen.Color.ToArgb());
+            if (DoubleBuffered)
+            {
+                for (uint h = 0; h < aHeight; h++)
+                {
+                    for (uint w = 0; w < aWidth; w++)
+                    {
+                        _xSVGADriver.SetPixel((uint)(w + aX_start), (uint)(aY_start + h), (uint)aPen.Color.ToArgb());
```

Comment:
Fill should also work with the double buffered implementation, so we dont have to handle it any differently and can just call `Fill` as well.